api/v1alpha1/virtualmachine_types.go

```diff
@@ -102,9 +102,40 @@ type VirtualMachineVolume struct {
 	// have a unique name.
 	Name string `json:"name"`
 
+	// Storage Policy Based Management (SPBM) profile ID associated with the StoragePolicyName.
+	// This currently only applies to volumes with VsphereVolumeSource.
+	// Volumes with PersistentVolumeClaimVolumeSource have their storage policy ID specified in the
+	// PersistentVolume spec.
+	// +optional
+	StoragePolicyID   *string `json:"storagePolicyID,omitempty"`
+
+	// Storage Policy Based Management (SPBM) profile name.
+	// This currently only applies to volumes with VsphereVolumeSource.
+	// Volumes with PersistentVolumeClaimVolumeSource have their storage policy name specified in the
+	// PersistentVolume spec.
+	// +optional
+	StoragePolicyName   *string `json:"storagePolicyName,omitempty"`
+
 	// PersistentVolumeClaim represents a reference to a PersistentVolumeClaim in the same namespace. The PersistentVolumeClaim
 	// must match a persistent volume provisioned (either statically or dynamically) by the cluster's CSI provider.
+	// +optional
 	PersistentVolumeClaim *corev1.PersistentVolumeClaimVolumeSource `json:"persistentVolumeClaim,omitempty"`
+
+	// VSphereVolume represents a reference to a vSphereVolumeSpec in the same namespace. Only one of PersistentVolumeClaim or
```

Comment:
- Comment should be `VsphereVolumeSource` instead of `vSphereVolumeSpec`
- The mutual exclusive nature of `PersistentVolumeClaim` or `VSphereVolume` should be commented as part of the `VirtualMachineVolume struct` comment.

Reply:
Done!